ts/DataGrid/DataGrid2/Actions/ColumnsResizer.ts

```diff
@@ -0,0 +1,250 @@
+/* *
+ *
+ *  Data Grid Columns Resizer class.
+ *
+ *  (c) 2020-2024 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Dawid Dragula
+ *  - Sebastian Bochan
+ *
+ * */
+
+'use strict';
+
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import DataGridTable from '../DataGridTable.js';
+import DataGridColumn from '../DataGridColumn.js';
+
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The class that handles the resizing of columns in the data grid.
+ */
+class ColumnsResizer {
+
+    /* *
+    *
+    *  Properties
+    *
+    * */
+
+    /**
+     * The minimum width of a column.
+     */
+    private static readonly MIN_COLUMN_WIDTH = 20;
+
+
+    /**
+     * The viewport of the data grid.
+     */
+    private viewport: DataGridTable;
+
+    /**
+     * The column being dragged.
+     */
+    private draggedColumn?: DataGridColumn;
+
+    /**
+     * The column right next to the dragged column.
+     */
+    private nextColumn?: DataGridColumn;
```

Comment:
Do we need it? The next column is `columns[index + 1]`, especially that we have an index of the current one.